docs(chart): drilldown chart documentation (#1592)
telerik/blazor-docs: components/chart/drilldown.md

---
title: DrillDown

> page_title: DrillDown 

Reviewer: Add the component name here too.

```suggestion
page_title: DrillDown 
```

Author: This is basically a functionality of the component, not a different chart type. We've discussed it and decided to remove the name of the component.

https://github.com/telerik/blazor-docs/pull/1592#discussion_r1301127062

In our other articles related to chart functionalities, we do not include the component name. Should we add it?

description: The DrillDown Chart feature allows the users to explore hierarchical data by initially displaying summarized information and to "drill down" into specific categories or data points for more detailed insights.
slug: chart-drilldown
tags: telerik,blazor,chart,drill,down,drilldown
published: true
position: 55
---

# DrillDown Charts

The Telerik UI for Blazor Chart supports drilldown functionality for exploring data.

> The drill-down feature allows the users to click on a point (for example, bar, pie segment, and more) to navigate to a different view that contains finer-grained data about the selected data point, like breakdown by product of the selected category. The view hierarchy can be displayed in a [Breadcrumb]({%slug breadcrumb-overview%}) for easy navigation back to previous views.

Reviewer: ```suggestion
The drill-down feature allows the users to click on a data point (for example, bar, column, pie segment, etc.) to navigate to a different view that contains finer-grained data like breakdown by product of the selected category. The view hierarchy can be displayed in a [Breadcrumb]({%slug breadcrumb-overview%}) for easy navigation back to previous views.
```

## Configuring DrillDown Charts

To configure Chart series for drill-down:

1. Prepare the data in the appropriate format. Each series data that will be drilled-down must contain a property of type `ChartSeriesDescriptor`. The descriptor includes all the parameters of the `ChartSeries` tag and acts as a container holding information about the series displayed upon user-initiated drill-down.
1. Specify the drilldown field (the `ChartSeriesDescriptor` field) of the series data by using the `ChartSeries.DrilldownField` or `ChartSeriesDescriptor.DrilldownField` property.

````CSHTML
@* Configuring the DrillDown Chart *@

<TelerikChart>
    <ChartSeriesItems>
        <ChartSeries Type="ChartSeriesType.Column"
                     Name="Total Sales By Company"
                     Data="@Data"
                     Field="@nameof(CompanyModel.Sales)"
                     CategoryField="@nameof(CompanyModel.Name)"
                     DrilldownField="@nameof(CompanyModel.Details)">
        </ChartSeries>
    </ChartSeriesItems>
</TelerikChart>

@code {

>     public List<CompanyModel> Data { get; set; } = new List<CompanyModel>();

Reviewer: Use private accessors.
The data can be hard-coded and defined outside OnInitialized. Or, switch the places of the two methods, so that the more important code is on top.

---

> title: Drilldown Charts

Reviewer 2: I would suggest removing "Charts" and keeping only "DrillDown" in the documentation navigation tree, since none of the titles contains "Charts".

page_title: Drilldown Charts
description: The drilldown Chart feature allows users to explore hierarchical data by initially displaying summarized information and enabling users to "drill down" into specific categories or data points for more detailed insights.
slug: chart-drilldown
tags: telerik,blazor,chart,drill,down,drilldown
published: true
position: 55
---

# DrillDown Charts

The Telerik UI Chart supports drilldown functionality for exploring data.

> The drilldown function allows users to click on a point (bar, pie segment, etc.) in order to navigate to a different view. The new view usually contains finer grained data about the selected item, like breakdown by product of the selected category.

Reviewer 2: Link the "Breadcrumb" with its Overview article.

> The drill-down feature allows the users to click on a point (for example, bar, pie segment, and more) to navigate to a different view that contains finer-grained data about the selected data point, like breakdown by product of the selected category. The view hierarchy can be displayed in a Breadcrumb for easier navigation back to previous views.

Reviewer 2: ```suggestion
The drill-down feature allows the users to click on a point (for example, bar, pie segment, and more) to navigate to a different view that contains finer-grained data about the selected data point, like breakdown by product of the selected category. The view hierarchy can be displayed in a [Breadcrumb]({%slug breadcrumb-overview%}) for easier navigation back to previous views.
```

> The view hierarchy can be displayed in a [breadcrumb]({%slug breadcrumb-overview%}) for easy navigation back to previous views.

Reviewer 2: Remove this sentence, since it is now added to the previous paragraph.

## Configuring Drilldown Charts

To configure a chart series for drilldown:

> 1. Prepare the data in the appropriate format. Each series data that will be drilled down should contain a property of type `ChartSeriesDescriptor`. The descriptor includes all the parameters of the `ChartSeries` tag and acts as a container holding information about the series displayed upon user-initiated drilldown.

Reviewer 2: ```suggestion
1. Prepare the data in the appropriate format. Each series data that will be drilled-down must contain a property of type `ChartSeriesDescriptor`. The descriptor includes all the parameters of the `ChartSeries` tag and acts as a container holding information about the series displayed upon user-initiated drill-down.
```

Reviewer 2: Please provide more information on the required data format.
